### ISSUE TITLE
[일지] Deep Learning 파트 5/3 일지

### ISSUE THREAD

**Issue by changwoomon** [boostcamp-ai-tech-4/ai-tech-interview#166]
### Discussed in https://github.com/boostcamp-ai-tech-4/ai-tech-interview/discussions/70

<div type='discussions-op-text'>

<sup>Originally posted by **bsm8734** May  5, 2021</sup>
## 날짜 및 시간
- 2021.05.03. (일) 15:00 ~ 18:00

## 장소
- 강남역 스터디카페(허슬버슬)

## 분야
- Deep-Learning

## 참석자
- [샐리🐤](@bsm8734)
- [원딜🌸](@changwoomon)
- [펭귄🐧](@CoodingPenguin)
- [히스👼🏻](@Heeseok-Jeong)

## 지각비
- [샐리🐤](@bsm8734) 1000원 (✔︎완료)
- [원딜🌸](@changwoomon) 2000원 (✔︎완료)

### [샐리🐤](@bsm8734) 님 파트

- 1 
  - "주로 정형데이터는 머신러닝, 비정형데이터는 딥러닝 쪽" 언급
- 4
  - block latex `$$ $$`으로 변경(식 크기 이상)
  - weight decay이 parameter norm penalty와 같은 것 → 수정
- 7
  - 선험적 지식: 경험하지 않아도 알 수 있는 것
  - 휴리스틱: 체계적이면서 합리적인 판단이 굳이 필요하지 않은 상황에서 사람들이 빠르게 사용할 수 있게 보다 용이하게 구성된 간편추론의 방법
    - '대충 어림짐작하기', '눈대중으로 맞추기'
  - "weight는 학습 과정에서 바뀌는 값들" → 추가
- 10
  - 인용구로 사용하지 말고 바로 답변으로 추가 (인용구 삭제하고 내용 합치기)
  - (+) ide와 idle 질문: IDE는 편집 툴, IDLE은 python interactive command tool과 같은 것
- 14
  - 아래의 추가 설명을 위로 올리기 (local minima 설명 위로 올리는 것이 좋을 것 같음)
  - 고차원의 공간에서는 local minima 문제가 발생하기 힘들다고 한다.
  - 변곡점/안장점 추가
- 14-1
  - NAG 부분에서 "미리 한스텝 가본다"는 내용 추가
  - SGD에 대한 조금이라도 설명 달기
- 14-2
  - 오타 수정
  - 조금 더 논리적으로 정리하기
- 18, 18-1, 18-2, 18-3
  - `파이썬 머신러닝 완벽가이드`를 `밑바닥 딥러닝`으로 출처 수정

### [원딜🌸](@changwoomon) 님 파트

- 3
  - 텐서플로우는 `y_pred, diff, loss` 모두를 직접 다 정의를 해줘야 함
  - 파이토치는 `y_pred, diff, loss`를 새로 만들어가면서 함
  - 직관적이고 편리해서 pytorch로 많이 넘어감
- 6
  - drop out 앙상블 효과가 있음에 대한 내용 추가 → "학습할때 껐다가 켰다가를 반복하므로"
  - label smoothing 그림이 augmentation에 대한 내용같아서 수정 필요
- 9
  - 볼츠만 머신의 역할
  - 색이 뭘 뜻하는지 써주면 좋을 것 같음
  - 노랑: input, 녹색: hidden
- 13-2
  - minima → global/minima
- 13-3
  - “역전파 알고리즘은 출력값에 대한 입력값의 기울기(미분값)을 출력층 layer에서부터 계산하여 거꾸로 전파시키는 것이다.이렇게 거꾸로 전파시켜서 최종적으로 출력층에서의 output값에 대한 입력층에서의 input data의 기울기 값을 구할 수 있다”
  - 문장 수정(출력이 아닌, Loss로 부터 역전파를 구하는 게 맞다고 생각)
- 17
  - SGD랑 미니배치를 분리 or 포함관계임을 적어주기
  - 식 중간에 줄바꿈 수정
- 17-3
  - 식 띄어쓰기
 
### [펭귄🐧](@CoodingPenguin) 님 파트

- 12
  - relu bias에 대한 내용도 있으면 좋을듯
- 12-1
  - 선형대수 관점
    - 더 했을때, 곱했을 때 유지되어야하는 속성
    - "덧셈에 대해 닫혀있다"→ 사진 참고
    <img src="https://user-images.githubusercontent.com/35002768/117030744-e49af100-ad3a-11eb-9c3d-33d293dcc50e.png" width="40%">
- 12-2
  - "zero center 문제를 해결하는 것이 왜 중요한가"에 대한 내용은 cs231n 6강을 참고하기
  - Relu는 zero centered 가 아니라는 문제가 있는데, 이 문제를 batch norm이 해결할 수 있음
  - zero centered 를 만족하지 못할 때 생기는 문제
    - zigzag가 생겨서 학습 속도가 느려짐
    - `펭귄`님 필기 첨부
       <img src="https://user-images.githubusercontent.com/35002768/117031012-23c94200-ad3b-11eb-84ce-127f1f2ec4de.png" width="50%">
- 12-3
  - 가중치만 조절하는 것은 기울기만 수정하는 것과 같음
- 16
  - 대문에 → 때문에
- 16-2
  - 왜 BN을 test에 사용할 떄 train의 평균, 분산을 적용할까?
- 16-3
  - GAN의 generater의 hidden layer에는 있을수도 있지 않을까? → 한번 더 찾아보기!
- 20-1
  - GPU 자체를 못 쓰는 케이스 추가
  - 배치를 적게 쓰고있기 때문에 배치를 늘려야한다는 내용 추가
- 20-3
  - 모델의 크기 = weight의 수

### [히스👼🏻](@Heeseok-Jeong) 님 파트

- 2 
  - activation에 대한 설명 추가
- 5
  - zero center 문제에 대한 설명을 펭귄님 문제에서 가져오기! → 링크
- 8
  - `Xavier Initialization LeCun 방법과 비슷하지만 들어오는 노드 수와 나가는 노드 수에 의존하고, 적절한 상수값도 발견하여 사용한 방법이다.` 부분에서 xavier intialization 뒤에 줄바꿈 필요
  - lecun에서 n_in 들어오는 layer 관련 이야기 추가(들어오는 레이어의 개수만 따른다는 설명)
- 11
  - one shot이 pretrain, 전이학습을 사용하는 것임
- 15-3
  - `원딜`님 질문이랑 중복 → 링크 필요
    - [#6 오버피팅일 경우 어떻게 대처해야 할까요?](https://github.com/boostcamp-ai-tech-4/ai-tech-interview/blob/main/answers/deep-learning.md#6)
- 19
  - receptive field 를 통해 이미지의 위치 정보까지 고려
- 19-3
  - 학습은 bce로 하되 성능을 mse를 쓴다는 말인가 싶음 (validation을 할 때)
  - criterion으로 bce를 쓰고, 성능을 mse로 본다는 질문이라고 볼 수 있을듯

----</div>
